### PR TITLE
Authentication and jwt token

### PR DESCRIPTION
**Pull request by BohdanZorii**

### CODE REVIEW

File: src/main/java/mate/zorii/bookstore/security/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/mate/zorii/bookstore/security/JwtAuthenticationFilter.java:43]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/mate/zorii/bookstore/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/zorii/bookstore/security/JwtAuthenticationFilter.java:44]: create constant for "Bearer "

File: src/main/java/mate/zorii/bookstore/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/zorii/bookstore/security/JwtAuthenticationFilter.java:45]: constant.length() (see comment above)

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve it

**Comment by BohdanZorii** [comment]
weird staff happened to my branches, but I fixed commened suggestions

**Review by MentorMateAcademy — APPROVED** [review]

**Pull request merged**